src/main/java/com/tterrag/k9/commands/CommandCurseGradle.java

```diff
@@ -234,54 +240,86 @@ private void downloadLibraries(String urlRead, String splitterText, String guiDi
+            }
+        }
+        
+        
+        editWaitMessage(waitMsg, guiMessage + ". Page " + page, ctx);
+        
+        Document urlRead = getDocument(projectURL + "/files?page=" + page);
+        Elements pageElement = urlRead.select("span.b-pagination-item").select("span.s-active");
+        if(!pageElement.isEmpty()) {
+            try {
+                if(Integer.valueOf(pageElement.html()) < page) {
+                    return;
+                }
+            } catch (NumberFormatException e) {
+                log.error("Enable to parse value of {} to an int. Returning on page {}", pageElement.html(), page);
+            }
         }
-        String[] urlReadLibs = urlRead.split("<tr class=\"project-file-list-item\">");
-        for(int i = 1; i < urlReadLibs.length; i++) {
-            if(urlReadLibs[i].split("<span class=\"version-label\">")[1].split("</span>")[0].equals(MCVersion)) {
-                calculate("https://minecraft.curseforge.com" + urlReadLibs[i].split("<a class=\"overflow-tip twitch-link\" href=\"")[1].split("\"")[0], waitMsg, ctx, list);
+
+        Elements libElements = urlRead.select("tr.project-file-list-item");
+
+        for(Element element : libElements) {
+            boolean isCorrectVersion = MCVersion.equalsIgnoreCase(element.select("span.version-label").html()); 
+            if(!isCorrectVersion) {
+                for(String version : element.select("span.additional-versions").attr("title").replace("<div>", "").split("</div>")) { //Get the list of extra hidden versions
+                    if(MCVersion.equalsIgnoreCase(version)) {
+                        isCorrectVersion = true;
+                        break;
+                    }
+                }
+
+            }
+            if(isCorrectVersion) {
+                calculate("https://minecraft.curseforge.com" + element.select("a.twitch-link").attr("href"), waitMsg, ctx, list);
                 return;
             }
         }
-        addLatestToList(projectURL, MCVersion, waitMsg, ctx, list, page++);
+        
+        addLatestToList(projectURL, MCVersion, waitMsg, guiMessage, ctx, list, page+=1);
     }
     
-    /**
-     * Used to get the data a URL holds.
-     * @param url The url to uses
-     * @param simulate Should the program <u>actually</u> download the file
-     * @return the data that the url points to. Usally a webpage
-     * @throws CommandException 
-     */
-    private String readURL(String url, boolean simulate) throws CommandException {
-        try {
-            InputStream urlStream = new URL(url).openStream();
-            String urlRead = "";
-            int len = urlStream.read();
-            if(simulate) {
-                while(len != -1) {
-                    urlRead += (char)len;
-                    len = urlStream.read();
-                }
+    private Document getDocument(String url) throws CommandException {
+        Document ret = null;
+        while (ret == null) {
+            try {
```

Comment:
Seems like this is copied from CommandCurse, could it be extracted?

Reply:
Do you mean using CommandCurses method?

Reply:
Or have a helper somewhere, or have them inherit from a superclass.

Reply:
Ill add the method to CommandBase, and have it inherit?